src/wsproxy.ml

```diff
@@ -45,7 +46,15 @@ let start path handler =
          Lwt_unix.close fd_sock2 >>= fun _ ->
```

Comment:
This is not the best code for this, and indeed it's hard to reason about, but the function is fine in this case. It is surrounded by safe functions (if they fail we have much bigger problems) and takes care of closing the fd on error and on success.

As an aside, I believe that this cli should need a great deal of refactoring but this would also require more time and we could plan it as a `wsproxy 2.0` release (that uses upstream base64 and upstream ocaml-websockets for example... reducing the maintenance burden)